### PR TITLE
Eslint prettier config

### PR DESCRIPTION
**Pull request by AdrianValdes**
Hi @paschalidi, Here is my first attempt to accomplish what you asked. Right now eslint and prettier are added to the whole project and we are able to use the command `npm run lint` on both the `ui` and the `fe-small-advertisments` folders.

Nevertheless, I think there is maybe a better way that doesn't involve installing the same devDependencies  three times, but I am not sure, I have some basic but important questions about how all this work. 
To do:
Solve the following eslint mistakes on the `ui` project:
1. `'@storybook/react' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies`
2. `Insert `⏎······``
3. `The button type attribute must be specified by a static string or a trivial ternary expression`

If I took the wrong path let me know and I will keep trying to find a better solution.

### CODE REVIEW

**Comment by AdrianValdes** [comment]
Do you have know any good documentation I could read about dependencies and node_modules, and how they connect to one or different projects?

File: packages/eslint-config-small-ads/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "name": "eslint-config-small-ads",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/package.json:11]: I think this is a good opotunity to discuss the difference between `devDependencies` and `dependencies`. Can you tell me whats the difference?

**Comment by AdrianValdes** [comment]
About avoiding installing the `devDependencies` for each folder, now I don't think it is possible, I found this line in the lerna official documentation:

> Note that devDependencies providing "binary" executables that are used by npm scripts still need to be installed directly in each package where they're used.

File: packages/eslint-config-small-ads/package.json (hunk 1 above)
Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:11]: The `dependencies` are require for the project to run, and therefore one should try to keep them small. The `devDependencies` are only required to develop the project. They can be big because they don't go in the build

Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:11]: The only dependency that I think it was weird to include in the `devDependencies` was `"typescript": "^4.1.3"`. But I got errors if I don't do it.

Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:11]: I am watching [this](https://www.youtube.com/watch?v=ZSdCNf1ncOE) conference on lerna now.

Comment (paschalidi) [reply packages/eslint-config-small-ads/package.json:11]: Super you know what they are. Now think _only_ about the `eslint-config-small-ads` package. And think of it as its own independent app. 

Which are the required dependencies for this project? Or in other words, which dependencies does the `eslint-config-small-ads` has the responsibility to install so that it operates correctly when we are to reuse it in some project?

Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:11]: Hmm, well if I focus only on eslint-config-small-ads I would install only the general devDependencies, those that could be reusable in whatever other project. The thing is, we know we want to use Typescript in this project, so I would include also those, and I would leave out the ones related to react.
"devDependencies": { "@typescript-eslint/eslint-plugin": "^4.14.0", "@typescript-eslint/parser": "^4.14.0", "eslint": "^7.18.0", "eslint-config-airbnb": "^18.2.1", "eslint-config-prettier": "^7.2.0", "eslint-import-resolver-typescript": "^2.3.0", "eslint-plugin-import": "^2.22.1", "eslint-plugin-prettier": "^3.3.1", "prettier": "^2.2.1" }

Comment (paschalidi) [reply packages/eslint-config-small-ads/package.json:11]: use ``` instead of one ` ;)

Comment (paschalidi) [reply packages/eslint-config-small-ads/package.json:11]: What you say isnt wrong. Also you are right including typescript and react related dependencies here. Perhaps not typescript itself but the eslint typescript/react related dependencies we want. 

Let me rephrase and be more specific now. 
1. We want this eslint config to be used by react projects that use typesript. 
2. Also we want to get straight, which dependencies are `devDependencies` and which are simply `dependencies`.

Lets focus our attention on that second requirement now. You are saying that dependencies are required for the project to run and devDependencies are used only in development. For example in a react project devDependency could be some testing library because when we build the react app and we make it a production executable that runs on the browser the testing library will not be necessary. 

Now I want us to think of the same way about this package. In the future we will install this package in some react project. For this project to run some dependencies are necessary and some dependencies are not necessary. 
Which would be necessary and which not?

Comment (paschalidi) [reply packages/eslint-config-small-ads/package.json:11]: I will let you think about it :) It is a riddle because I dont wanna give you the answer. I think you will benefit more understanding this by doing your own research

Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:11]: Ooooh, I think I go it! Shit, I was installing the packages as devDependencies, and because of that, when I was running `npm i ../packages/eslint-config-small-ads` on the other projects nothing was being installed.

Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:11]: But maybe I am wrong, I thought of that while taking a shower :D I didn't read anything new I just thought about it. So maybe I have to keep thinking and researching

Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:11]: Unfortunately Maria and I had the plan of watching a movie. Now I am being forced to "relax" I cannot keeping studying, boomer :(.

Comment (paschalidi) [reply packages/eslint-config-small-ads/package.json:11]: Great great! 

RELAX. I also need to ;) We do that tomorrow

Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:11]: Good Morning @paschalidi, I changed the `devDependencies` and made them `dependencies` in the `eslint-config-small-ads` project, but it is still not working. 
I installed `eslint-config-small-ads` in `fe-small-advertisements` but still looks as if I have to install everything manually.

Comment (paschalidi) [reply packages/eslint-config-small-ads/package.json]: Well, there is a reason for that. I neglected to mention, that while you work with lerna, we dont navigate into each project and manually doing npm i for each one of them. What we are doing is basically asking lerna to do this work for us. This way it happens in parallel for all the different projects lerna is responsible for. 

What you want to do is read these commands [here](https://github.com/lerna/lerna). Lerna bootstrap specifically.

**Comment by AdrianValdes** [comment]
In the `eslint-config-small-ads` project, I also add the `dependencies` as `peerDependencies` as it says here[https://eslint.org/docs/developer-guide/shareable-configs#creating-a-shareable-config](url) and try to link the project with `npm link.`

File: packages/ui/package.json [hunk 2]
```diff
@@ -48,6 +49,17 @@
...
+    "@typescript-eslint/parser": "^4.14.0",
+    "eslint": "^7.18.0",
+    "eslint-config-airbnb": "^18.2.1",
+    "eslint-config-airbnb-typescript": "^12.0.0",
+    "eslint-config-prettier": "^7.2.0",
+    "eslint-config-small-ads": "file:../../packages/eslint-config-small-ads",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-prettier": "^3.3.1",
+    "prettier": "^2.2.1"
```
Comment (paschalidi) [inline packages/ui/package.json:63]:
```suggestion
    "eslint-config-small-ads": "^1.0.0"
```

Basically you will end up with having here only this. When you do `lerna bootstrap` lerna will see that this package exists in the repository and it will connect it for us without it being published in npm or anything.

File: packages/eslint-config-small-ads/package.json (hunk 1 above)
Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json]: Oh, I just saw this, let me see

**Comment by paschalidi** [comment]
I am a bit busy today. Please keep it up till the evening when we meet. Lets say we meet today at 2030 berlin time online :)

**Comment by AdrianValdes** [comment]
hi @paschalidi, Now the `eslint-config-small-ads` is linked to both the `ui` and the `fe-small-advertisements` projects. I am not getting errors and I am able to open run `npm start` in both. 

Nevertheless, I am not being able to run `npm run lint` from the `root`, I am getting the error:

```
> root@ lint /home/dci/learningToProgram/Web-Development/projects/with-christos/core
> lerna run eslint

lerna notice cli v3.22.1
lerna info Executing command in 2 packages: "npm run eslint"
lerna ERR! npm run eslint exited 1 in '@small-advertisements/ui'
lerna ERR! npm run eslint stdout:

> @small-advertisements/ui@0.1.0 eslint /home/dci/learningToProgram/Web-Development/projects/with-christos/core/packages/ui
> src --ext .ts,.tsx --config ./.eslintrc.js


lerna ERR! npm run eslint stderr:
sh: 1: src: not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! @small-advertisements/ui@0.1.0 eslint: `src --ext .ts,.tsx --config ./.eslintrc.js`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the @small-advertisements/ui@0.1.0 eslint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dci/.npm/_logs/2021-01-26T10_23_44_650Z-debug.log

lerna ERR! npm run eslint exited 1 in '@small-advertisements/ui'
lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! root@ lint: `lerna run eslint`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the root@ lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dci/.npm/_logs/2021-01-26T10_23_44_677Z-debug.log
```
I am trying to solve it, but I find weird answers on SlackOverflow. Supposedly I have to run `npm cache clean`, but I don't want to do it yet because I am not sure what to do. I know you are busy so I still won't ask for a review, I will try to solve it myself.

**Comment by AdrianValdes** [comment]
I solved the issue, now `lerna` handles the running of `eslint` for each package.

**Comment by AdrianValdes** [comment]
I know you are busy until Thursday, no need to rush. I am not getting errors but I am scared something can come up : ), well I am not really scared, I understand all this is necessary, I really do.

Either way, I need to study Typescript. I will do that in the meantime.

File: packages/eslint-config-small-ads/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    ecmaFeatures: {
+      jsx: true,
+    },
+    ecmaVersion: 12,
+    sourceType: "module",
+  },
+  plugins: ["react-hooks", "@typescript-eslint/eslint-plugin"],
+  rules: {
+    "import/prefer-default-export": "off",
+    "import/no-extraneous-dependencies": "off",
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/index.js:26]: You dont want to deactivate this for _all_ our packages. You can do that extending the rules only for the ui package. Remember the princile [Open-Closed Principle](https://www.digitalocean.com/community/conceptual_articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design#open-closed-principle). Modules should be extendable without having to manipulate the module itself.

File: packages/eslint-config-small-ads/index.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    },
+  },
+  overrides: [
+    {
+      files: ["*.ts", "*.tsx"],
+      parserOptions: {
+        project: ["tsconfig.json"],
+      },
+    },
+  ],
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/index.js:42]: You want this to be be in each module we are having and not be living in the eslint package. 

The eslint package is only about eslint configuration we dont wanna be having the tsconfig file in this package but each module should be responsible for their own tsconfig. This way if we wanted the `ui` to have different `tsconfig` from the `fe` package we could be doing that.

File: packages/eslint-config-small-ads/package.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    "eslint-config-airbnb": "^18.2.1",
+    "eslint-config-prettier": "^7.2.0",
+    "eslint-import-resolver-typescript": "^2.3.0",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-prettier": "^3.3.1",
+    "eslint-plugin-react": "^7.22.0",
+    "eslint-plugin-react-hooks": "^4.2.0",
+    "prettier": "^2.2.1"
+  }
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/package.json:40]: I dont think we need those, do we?

File: packages/eslint-config-small-ads/package.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    "eslint-config-airbnb-typescript": "^12.0.0",
+    "eslint-config-prettier": "^7.2.0",
+    "eslint-config-react": "^1.1.7",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-prettier": "^3.3.1",
+    "eslint-plugin-react": "^7.22.0",
+    "eslint-plugin-react-hooks": "^4.2.0",
+    "prettier": "^2.2.1",
+    "typescript": "^4.1.3"
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/package.json:25]: Also if we move tsconfig to each packahe I think typescript isnt needed as a dependency here.

File: packages/eslint-config-small-ads/package.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "eslint-config-small-ads",
+  "version": "1.0.0",
+  "description": "",
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/package.json:4]: Mind writing a short description of that package here?

File: packages/eslint-config-small-ads/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,41 @@
+{
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/tsconfig.json:1]: As I see it this while file can be deleted. We will be having a tsconfig for each package.

File: packages/ui/.eslintrc.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  extends: ["eslint-config-small-ads"],
```
Comment (paschalidi) [inline packages/ui/.eslintrc.js:2]:
```suggestion
  extends: ["eslint-config-small-ads"],
  rules: {
    "import/no-extraneous-dependencies": "off",
  }
```

Comment (paschalidi) [reply packages/ui/.eslintrc.js:2]: Here is how we can extend the rules for a package if we wanted it. This is why I am mentioning this rule should be here only and not in the eslint package.

File: packages/ui/package.json [hunk 10]
```diff
@@ -22,7 +22,8 @@
   "scripts": {
     "start": "start-storybook -p 6006 -s public",
     "build": "build-storybook -s public",
-    "test": "react-scripts test"
+    "test": "react-scripts test",
+    "lint": "eslint '*/**/*.{js,ts,tsx}' --config ./.eslintrc.js"
```
Comment (paschalidi) [inline packages/ui/package.json:26]: Are we having any js files? Lets remove the `js` we can add it if we are having any js files in the `ui`. 
```suggestion
    "lint": "eslint '*/**/*.{ts,tsx}' --config ./.eslintrc.js"
```

File: packages/ui/.eslintrc.js (hunk 9 above)
Comment (paschalidi) [reply packages/ui/.eslintrc.js:2]: Also here is where we want this bit to come 
```
  overrides: [
    {
      files: ["*.ts", "*.tsx"],
      parserOptions: {
        project: ["tsconfig.json"],
      },
    },
  ],
```

And then we we can do the same for the `src/fe..`.

File: packages/eslint-config-small-ads/package.json (hunk 5 above)
Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:40]: True, we don't. They are gone.

File: packages/eslint-config-small-ads/package.json (hunk 7 above)
Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:4]: I tried to find best practices on line, there is not much info though. I tried to be short and declarative. But I will listen to any suggestion you may come up.

File: packages/eslint-config-small-ads/tsconfig.json (hunk 8 above)
Comment (AdrianValdes) [reply packages/eslint-config-small-ads/tsconfig.json:1]: Hmm, it is true. I deleted.

File: src/fe-small-advertisements/package.json [hunk 11]
```diff
@@ -19,7 +19,8 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test"
+    "test": "react-scripts test",
+    "lint": "eslint '*/**/*.{js,ts,tsx}' --config ./.eslintrc.js"
```
Comment (paschalidi) [inline src/fe-small-advertisements/package.json:23]: same as before

File: packages/ui/src/stories/Button/index.tsx [hunk 12]
```diff
@@ -1,10 +1,10 @@
...
 
 export interface ButtonProps {
-  type: "primary" | "secondary";
+  type: 'primary' | 'secondary';
   id: string;
-  htmlType: "button" | "submit" | "reset" | undefined;
+  htmlType: 'button' | 'submit' | 'reset' | undefined;
   onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
   disabled: boolean | undefined;
   style: object;
```
Comment (paschalidi) [inline packages/ui/src/stories/Button/index.tsx:10]: Thinking of the types again, most of them arent right here. I missed it on my initial review. Can you please make an issue for it so that we can work it out?

File: packages/ui/package.json [hunk 13]
```diff
@@ -22,7 +22,8 @@
...
   },
   "eslintConfig": {
```
Comment (paschalidi) [inline packages/ui/package.json:28]: We want to delete this section from both this one and the `fe`

**Review by paschalidi — COMMENTED** [review]
Looks better! Good work. I ve made some comments you might wanna be looking into. 

Also prettier seems not to be working unless is installed in the root directory, can we please install it there as well?

Also please if you having any other question I would be happy to answer them

File: packages/eslint-config-small-ads/.prettierrc [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+  "bracketSpacing": true,
+  "jsxBracketSameLine": false,
+  "jsxSingleQuote": false,
+  "printWidth": 80,
+  "proseWrap": "always",
+  "semi": false,
+  "singleQuote": false,
+  "tabWidth": 2,
+  "trailingComma": "all"
+}
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/.prettierrc:11]: After we install prettier as devDep on the root, can we try to delete this? I believe this isnt needed for now.

File: packages/eslint-config-small-ads/index.js (hunk 3 above)
Comment (AdrianValdes) [reply packages/eslint-config-small-ads/index.js:26]: I understand now, thanks for the explanation and the and for making me think about how to extend modules. I can see how it is better to have particular configuration for each project to their own `.eslintrc.json`.

File: packages/ui/src/Colors/index.ts [hunk 15]
```diff
@@ -1,20 +1,20 @@
 export const Colors = {
-  primary: "#0071f3",
```
Comment (paschalidi) [inline packages/ui/src/Colors/index.ts:2]: Looks odd that your linters are set up to fix this in single quote while mine are not.. We might need to check this together with a call at some point, somthing is off

File: packages/eslint-config-small-ads/index.js (hunk 4 above)
Comment (AdrianValdes) [reply packages/eslint-config-small-ads/index.js:42]: The same as above, I can see how it is better this way, thanks for pointing out. And for explaining it.

File: packages/eslint-config-small-ads/package.json (hunk 6 above)
Comment (AdrianValdes) [reply packages/eslint-config-small-ads/package.json:25]: If I delete Typescript I get this error whenever I try to run the projects.
```
Failed to load parser '@typescript-eslint/parser' declared in '.eslintrc.js » eslint-config-small-ads': Cannot find module 'typescript'
Require stack:
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/packages/eslint-config-small-ads/node_modules/@typescript-eslint/typescript-estree/dist/parser.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/packages/eslint-config-small-ads/node_modules/@typescript-eslint/typescript-estree/dist/index.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/packages/eslint-config-small-ads/node_modules/@typescript-eslint/parser/dist/parser.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/packages/eslint-config-small-ads/node_modules/@typescript-eslint/parser/dist/index.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/@eslint/eslintrc/lib/config-array-factory.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/@eslint/eslintrc/lib/index.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/eslint/lib/cli-engine/cli-engine.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/eslint/lib/cli-engine/index.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/eslint/lib/api.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/eslint-webpack-plugin/dist/getESLint.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/eslint-webpack-plugin/dist/linter.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/eslint-webpack-plugin/dist/index.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/eslint-webpack-plugin/dist/cjs.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/react-scripts/config/webpack.config.js
- /home/dci/learningToProgram/Web-Development/projects/with-christos/core/src/fe-small-advertisements/node_modules/react-scripts/scripts/start.js
```

File: tsconfig.json [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+        "moduleResolution": "node",
+        "resolveJsonModule": true,
+        "isolatedModules": true,
+        "noEmit": true,
+        "jsx": "react-jsx"
+    },
+    "include": [
+        "src"
+    ]
+}
```
Comment (paschalidi) [inline tsconfig.json:26]: I dont think we need this either..

File: packages/eslint-config-small-ads/package.json (hunk 6 above)
Comment (paschalidi) [reply packages/eslint-config-small-ads/package.json:25]: Will doublecheck.

File: packages/ui/src/Colors/index.ts (hunk 15 above)
Comment (AdrianValdes) [reply packages/ui/src/Colors/index.ts:2]: That is odd indeed, all mine are single quotes.

File: tsconfig.json (hunk 16 above)
Comment (AdrianValdes) [reply tsconfig.json:26]: We don't, they are being configured in every project already, it makes sense

File: packages/eslint-config-small-ads/package.json (hunk 6 above)
Comment (paschalidi) [reply packages/eslint-config-small-ads/package.json:25]: You right about this

File: packages/eslint-config-small-ads/.prettierrc (hunk 14 above)
Comment (AdrianValdes) [reply packages/eslint-config-small-ads/.prettierrc:11]: I thought we would need this to tell prettier how to behave. I mean, I thought eslint would use whatever prettier would advice whenever they would have clashing rules.

Comment (AdrianValdes) [reply packages/eslint-config-small-ads/.prettierrc:11]: But I installed in the root, and removed the file like you suggested. I suppose everything is being handle in the `index.js` from `eslint-config-small-ads` but it is still a bit confusing.

File: packages/ui/src/stories/Button/index.tsx (hunk 12 above)
Comment (AdrianValdes) [reply packages/ui/src/stories/Button/index.tsx:10]: I opened the issue. By the way, everything on my side is single quoted `htmlType: 'button' | 'submit' | 'reset' | undefined;`

File: packages/eslint-config-small-ads/.prettierrc (hunk 14 above)
Comment (paschalidi) [reply packages/eslint-config-small-ads/.prettierrc:11]: Lets get this right the next days. We can have a talk, but first lets figure out why we have different quotes.

**Comment by AdrianValdes** [comment]
I made all the requested changes, except for one. I couldn't remove `"typescript": "^4.1.3"` as a dependency from `eslint-config-small-ads`. 
I do have some questions:
1. I run `npm run bootstrap` in the root every time I made changes in the `packages.json`, but I am not sure if that is necessary.

2. I am also not super clear about the interconnection between `eslint` and `prettier`, I think I understand what they do, but I don't really understand how this: 
```
extends: [
    "plugin:import/recommended",
    "plugin:import/typescript",
    "plugin:react-hooks/recommended",
    "airbnb-typescript",
    "prettier",
    "prettier/react",
    "prettier/@typescript-eslint",
  ],
  ```
  would solve the problem exactly. I know it solves it, but I don't know HOW.
  
 My understanding was that because prettier comes after in the array it would be handling the auto-code-formatting. And because of this, we would need to have some kind of configuration in a `.prettierrc.json`. For instance, now we have `prettier` as a `devDependency` in the root, but we don't have `.prettierrc.json` to the prettier how to behave. You mentioned that 

> Looks odd that your linters are set up to fix this in single quote while mine are not.

But I thought  prettier would be in charge of that and not the linters.

**Comment by AdrianValdes** [comment]
I also need to read more about how this bit works:

```
overrides: [
    {
      files: ['*.ts', '*.tsx'],
      parserOptions: {
        project: ['tsconfig.json'],
      },
    },
  ],
  ```
  But I will do that alone for now : )

**Comment by paschalidi** [comment]
You might want to push your changes ;) 

> Looks odd that your linters are set up to fix this in single quote while mine are not.

It is prettier yeah, but sort to speak is a linter. It is not technically 100 percent correct term you right, but those things are almost the same. Its right that prettier is responsible for formatting the code, and simply takes the instructions from the eslint. 

Is having a `prettier.js` works for you?

**Comment by paschalidi** [comment]
> I run npm run bootstrap in the root every time I made changes in the packages.json, but I am not sure if that is necessary.

Perhaps is not, are the node_modules being updated on the fly on the when you change something on the `eslint-config` package?

**Comment by AdrianValdes** [comment]
> You might want to push your changes ;)
Shit I think I just committed and didn't push, I have to come to Github and double check always, which in my mind I did :(

**Comment by AdrianValdes** [comment]
> It is prettier yeah, but sort to speak is a linter. It is not technically 100 percent correct term you right, but those things are almost the same. Its right that prettier is responsible for formatting the code, and simply takes the instructions from the eslint.

I understand, I used the wrong terms because Prettier *is*, at the end, linting. So I have to rephrase and reorganize those terms in my mind to say them correctly. 

Again thanks for pointing that out and taking your time to do it. Don't think I will be annoyed, seriously! I won't think something like: "Oh, why is he correcting me? It is the same at the end." I really appreciate you pointing out every issue, no matter how small, because I do think they are important, and I do think those details matter. And I prefer to learn all that from someone that is close to me than later from someone else. So please continue doing it, I know is more work for you, I thank you for that.

File: packages/eslint-config-small-ads/index.js [hunk 17]
```diff
@@ -21,23 +21,11 @@ module.exports = {
     sourceType: "module",
   },
   plugins: ["react-hooks", "@typescript-eslint/eslint-plugin"],
-  rules: {
-    "import/prefer-default-export": "off",
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/index.js]: Hey, we want this one to be global _but_ the `"import/no-extraneous-dependencies": "off",` is specific to the `ui`

File: packages/eslint-config-small-ads/package.json [hunk 18]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "eslint-config-small-ads",
   "version": "1.0.0",
-  "description": "",
+  "description": "An application to add eslint-airbnb support, integrated with Typescript and Prettier",
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/package.json:4]: SUPER!

File: packages/eslint-config-small-ads/package.json [hunk 19]
```diff
@@ -24,18 +24,5 @@
...
-    "eslint-config-prettier": "^7.2.0",
-    "eslint-import-resolver-typescript": "^2.3.0",
-    "eslint-plugin-import": "^2.22.1",
-    "eslint-plugin-jsx-a11y": "^6.4.1",
-    "eslint-plugin-prettier": "^3.3.1",
-    "eslint-plugin-react": "^7.22.0",
-    "eslint-plugin-react-hooks": "^4.2.0",
-    "prettier": "^2.2.1"
-  }
+  "peerDependency": {}
```
Comment (paschalidi) [inline packages/eslint-config-small-ads/package.json:27]: 👍

File: packages/ui/package.json (hunk 13 above)
Comment (paschalidi) [reply packages/ui/package.json:27]: Have you had the time to look into deleting the `eslintConfig` section?

File: src/fe-small-advertisements/.eslintrc.js [hunk 20]
```diff
@@ -1,3 +1,14 @@
 module.exports = {
-  extends: ["eslint-config-small-ads"],
+  extends: ['eslint-config-small-ads'],
+  rules: {
+    'import/prefer-default-export': 'off',
```
Comment (paschalidi) [inline src/fe-small-advertisements/.eslintrc.js:4]: One thing, and this isnt something you could know of. But think of this situation like this. We want to have this rule for all our projects. Meaning exporting as constancts and not as default. Because of that we want this to be in the global `eslint` package, otherwise we will be repeating it each time. Now the one for the `ui` is unique to it and this is why we arent including it in the global eslint config. 

Do I explain that well?

File: src/fe-small-advertisements/package.json (hunk 11 above)
Comment (paschalidi) [reply src/fe-small-advertisements/package.json:22]: +1

File: package.json [hunk 21]
```diff
@@ -1,7 +1,12 @@
...
   "name": "root",
   "private": true,
+  "scripts": {
+    "bootstrap": "lerna bootstrap",
+    "lint": "lerna run lint"
+  },
   "devDependencies": {
-    "lerna": "^3.22.1"
+    "lerna": "^3.22.1",
+    "prettier": "^2.2.1"
```
Comment (paschalidi) [inline package.json:10]: Looks good!

File: packages/eslint-config-small-ads/index.js (hunk 17 above)
Comment (AdrianValdes) [reply packages/eslint-config-small-ads/index.js]: Oh, I saw I had to use it in both projects, but I wasn't not sure if we would need it in *every* project in case we create something new. But it makes sense that we have it global for now if we are using it in every project. Total sense.

File: packages/ui/src/Colors/index.ts (hunk 15 above)
Comment (paschalidi) [reply packages/ui/src/Colors/index.ts:2]: Lets chat about this situation here sometime, I will clone the repository fresh and install the dependencies and then run the `lint` command. This might give me some insights. I suggest you do the same 👍

File: src/fe-small-advertisements/package.json [hunk 22]
```diff
@@ -19,7 +19,7 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test"
+    "lint": "eslint '*/**/*.{js,ts,tsx}' --config ./.eslintrc.js"
   },
   "eslintConfig": {
```
Comment (paschalidi) [inline src/fe-small-advertisements/package.json:24]: Can you also delete this section from here?

**Review by paschalidi — COMMENTED** [review]
Last bits and we set to merge this! Check out my comments

File: packages/eslint-config-small-ads/index.js (hunk 17 above)
Comment (paschalidi) [reply packages/eslint-config-small-ads/index.js]: This is something it is in my head. And you couldnt really guess it :) So no stress, I should have explained myself better

File: src/fe-small-advertisements/.eslintrc.js (hunk 20 above)
Comment (AdrianValdes) [reply src/fe-small-advertisements/.eslintrc.js:4]: Yes, you did. Very well!

 I already got it when you said:

> Hey, we want this one to be global but the "import/no-extraneous-dependencies": "off", is specific to the ui.

That is why I said:

> Oh, I saw I had to use it in both projects, but I wasn't not sure if we would need it in every project in case we create something new. But it makes sense that we have it global for now if we are using it in every project. Total sense.

I pushed what I already committed yesterday, that is why this issue wasn't fixed yet. I didn't want to fix it in the previous commit to not make a mess, is that wrong? I mean, to push what I already committed before making more changes even though you made comments I could fix

Comment (AdrianValdes) [reply src/fe-small-advertisements/.eslintrc.js:4]: Anyhow, your explanation now was very clear and it draws a complete picture of the issue.

File: src/fe-small-advertisements/package.json (hunk 22 above)
Comment (AdrianValdes) [reply src/fe-small-advertisements/package.json]: Done

**Comment by AdrianValdes** [comment]
One question I have, are you also seeing an error from `eslint` when you open a file for the first time on the `ui` or on the `fe-small-advertisements`?
This is the error: 

> Parsing error: Cannot read file '/.../tsconfig.json'.eslint

Then if I open the `.tsconfig.json` for that project the error goes away.

I found [this](https://stackoverflow.com/questions/64933543/parsing-error-cannot-read-file-tsconfig-json-eslint) solution, but I want to be sure we are having the same errors before doing something, what do you think?

File: src/fe-small-advertisements/package.json (hunk 11 above)
Comment (paschalidi) [reply src/fe-small-advertisements/package.json:22]:
```suggestion
    "lint": "eslint '*/**/*.{ts,tsx}' --config ./.eslintrc.js"
```
@AdrianValdes check this out

**Comment by paschalidi** [comment]
Okay good news cloning the repo from skratch installing the dependencies and everything works. Pretties formats with me saving the file and eslint commant works as expected. 

Also singlequotes it is. Dunno why my previous clone didnt really feel like doing so :)

**Comment by paschalidi** [comment]
And yes there is the error, would be good if we can fix it with the solution you providing. Good catch!

**Comment by AdrianValdes** [comment]
I solved the issue and pushed it. 

On a aside annotation, I am checking my commit comments and I think I can improve them. I will read the Conventional Commits documentation again.

**Review by paschalidi — APPROVED** [review]
Lets get this in!!
